### PR TITLE
Add xlsx support

### LINKED ISSUE #274: [FEATURE] XLSX reports
I would like to propose the addition of XLSX report generation.
This feature aims to mirror the functionality of the current CSV reports. It will enable users to generate XLSX reports directly from Discover and Reports Dashboards

Implementation details: 
1. Download button should be added to popover:

2. The file format selector needs to be updated with xlsx format option

3. Reports Dashboard should allow to download xlsx format

4. Add xlsx format to opensearch-project/reporting

### PR DESCRIPTION
**Pull request by gulderov**
### Description
XLSX support added

This feature aims to improve the user experience by providing dynamic and visually appealing reports. Some enhancements are applied by default to make report more friendly to users

* Header row freeze
* Bold and aligned header text
* Auto filter for each column
* Column width adjusted by content

### Dependencies
This PR is part of improvements and should be merged in conjuction with opensearch-project/reporting#940

### Issues Resolved
Closes #274

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Please check report example:
[On_demand_report_2024-02-14T11_54_04.778Z_c073cca0-cb2f-11ee-b7b5-53678459d314.xlsx](https://github.com/opensearch-project/dashboards-reporting/files/14279214/On_demand_report_2024-02-14T11_54_04.778Z_c073cca0-cb2f-11ee-b7b5-53678459d314.xlsx)

1. Download button added to popover:
SCR-20240214-nlyp

2. The file format selector updated with xlsx format option:
SCR-20240214-nmnb

3. Reports Dashboard allowing to download xlsx format:
SCR-20240214-nmig

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,8 @@
     "react-toast-notifications": "^2.4.0",
     "set-interval-async": "1.0.33",
     "showdown": "^1.9.1",
-    "tesseract.js": "^4.0.2"
+    "tesseract.js": "^4.0.2",
+    "xlsx": "https://cdn.sheetjs.com/xlsx-0.20.1/xlsx-0.20.1.tgz"
```
Comment (joshuali925) [inline package.json:43]: looks like sheetjs is not publishing to [npm registry](https://www.npmjs.com/package/xlsx) anymore https://git.sheetjs.com/sheetjs/sheetjs/issues/3055, i'm not sure if our vulnerability scan tools support libraries like this

**Review by joshuali925 — COMMENTED** [review]
thanks for the PR! briefly went through looks good, i'll try to look more in detail

File: package.json (hunk 1 above)
Comment (wbeckler) [reply package.json:43]: Scanners scan built artifacts, and they identify some CVEs by examining package.json and subdeps. I have never heard of an npm requirement.

Comment (gulderov) [reply package.json]: @joshuali925 @wbeckler 

I have updated the pull request with transition from `xlsx` to `ExcelJS`

This change was motivated by several factors:
* Robust API
* Advanced styling options available out of the box
* Availability on npm

Comment (joshuali925) [reply package.json]: thanks. @sumukhswamy could you help to take a look at this when you can?

**Review by rupal-bq — APPROVED** [review]
LGTM

**Review by joshuali925 — APPROVED** [review]
thanks!

**Pull request merged**